Sources_v3/StreamChat/APIClient/APIClient.swift

```diff
@@ -63,4 +65,52 @@ class APIClient {
             task.resume()
         }
     }
+
+    func uploadFile<Response: Decodable>(
+        endpoint: Endpoint<Response>,
+        multipartFormData: MultipartFormData,
+        progress: ((Double) -> Void)? = nil,
+        completion: @escaping (Result<Response, Error>) -> Void
+    ) {
+        encoder.encodeRequest(for: endpoint) { [unowned self] (requestResult) in
+            var urlRequest: URLRequest
+            do {
+                urlRequest = try requestResult.get()
+            } catch {
+                log.error(error)
+                completion(.failure(error))
+                return
+            }
+
+            let data = multipartFormData.multipartFormData
+            urlRequest.addValue("multipart/form-data; boundary=\(multipartFormData.boundary)", forHTTPHeaderField: "Content-Type")
+            urlRequest.httpBody = data
+
+            let task = self.session.dataTask(with: urlRequest) { [decoder = self.decoder] (data, response, error) in
```

Comment:
With `uploadTask` it looks much better since we don't need explicitly create multipart data and set additional headers but was getting 413 error saying that the file exceeds the max size limit (it doesn't)